### PR TITLE
[launch-preflight] Trigger preflight on page load

### PR DESCRIPTION
**Pull request by wlonk**
But tests are failing.

### CODE REVIEW

File: src/js/components/plans/ctaButton.js [hunk 1]
```diff
@@ -97,6 +97,13 @@ class CtaButton extends React.Component<Props, { modalOpen: boolean }> {
     this.state = { modalOpen: false };
   }
 
+  componentDidMount() {
+    const { plan, doStartPreflight, user, preflight } = this.props;
+    if (user && user.valid_token_for && !(preflight && preflight.is_valid)) {
+      doStartPreflight(plan.id);
+    }
+  }
```
Comment (jgerigmeyer) [inline src/js/components/plans/ctaButton.js]: @wlonk My understanding from the Trello card is that we only want this to happen after clicking "Log In to Start Pre-Install Validation" button, not on every page-load. I think we talked about doing this with a query-string passed through the login flow?

**Comment by wlonk** [comment]
@jgerigmeyer ah, that sounds right. How do you propose the query string to work?

**Comment by jgerigmeyer** [comment]
@wlonk Tests will still fail here (I haven't touched them yet), but in the meantime I'd appreciate some eyes on the code (and functionality too).

File: src/js/components/plans/ctaButton.js [hunk 2]
```diff
@@ -91,10 +96,65 @@ export const ActionBtn = ({
...
+      // `preflight === null`: no prior preflight exists
+      // `preflight === undefined`: still fetching prior preflights
+      // If we don't know about past preflights yet, wait until we do...
+      if (preflight === undefined) {
+        this.setState({ startPreflight: true });
+      } else {
+        this.autoStartPreflight();
+      }
+      // Remove query-string from URL
+      history.replace({ search: '' });
```
Comment (wlonk) [inline src/js/components/plans/ctaButton.js]: Does this generate a back-button entry? Or would that only be on `pushState`?

File: src/js/components/plans/ctaButton.js [hunk 3]
```diff
@@ -91,10 +96,65 @@ export const ActionBtn = ({
...
+    const preflightChanged = preflight !== prevProps.preflight;
+    if (startPreflight && preflightChanged && preflight !== undefined) {
+      this.autoStartPreflight();
+    }
+  }
+
+  autoStartPreflight(): void {
+    const { plan, doStartPreflight } = this.props;
+    this.setState({ startPreflight: false });
+    if (this.canStartPreflight()) {
```
Comment (wlonk) [inline src/js/components/plans/ctaButton.js]: Ugh so much cleaner than what I wrote, a small well-named function that describes what it means, thank you!

File: src/js/components/plans/ctaButton.js (hunk 2 above)
Comment (jgerigmeyer) [reply src/js/components/plans/ctaButton.js]: This replaces the current entry. `history.push()` would create a new entry.

**Pull request merged**